roles/setup_lvms/tasks/main.yml

```diff
@@ -0,0 +1,22 @@
+---
+- name: Validate required variables
+  ansible.builtin.assert:
+    that:
+      - sl_dev_class_name | length > 0
+      - sl_dev_selector_paths | length > 0
+    fail_msg: "Required variables sl_dev_class_name and sl_dev_selector_paths must be defined"
+
+- name: Validate optional variables
+  ansible.builtin.assert:
+    that:
+      - sl_storage_class | regex_search('^[a-z0-9.-]+$')
+    fail_msg: "Optional variable sl_storage_class must contain any of the following characters a-z0-9.-"
+  when: sl_storage_class | length > 0
+
+- name: Include setup tasks
+  ansible.builtin.include_tasks: setup.yml
```

Comment:
For arg_specs to work here, you need to call the role and override the entry point.